### DISCUSSION TITLE
【添加应用】添加免费VPN服务器程序——萤火虫

### DISCUSSION THREAD

**Discussion by Safe3** [1Panel-dev/1Panel#4552] [Ideas]
### 1Panel 版本

all

### 请描述您的需求或者改进建议

项目地址：https://github.com/Safe3/firefly

软件介绍：
最简单易用的WireGuard服务器之一。内网穿透、异地组网、远程办公。

界面展示：
![image](https://raw.githubusercontent.com/Safe3/firefly/main/firefly.png)

**Comment by kmou424** [comment]
不建议上架，开发者的软件前端部分完全基于[wg-easy/wg-easy](https://github.com/wg-easy/wg-easy)，查看源码可以发现很多代码，甚至连注释都是一模一样的。 以下是部分证据。

原项目截取的某段注释：
https://github.com/wg-easy/wg-easy/blob/c2482f494ae9c7e1acd21db6ab04ee3e91d042da/src/www/index.html#L381-L393
开发者项目中截取的相同注释：
https://github.com/Safe3/firefly/blob/47da61c0afbfa4232bbde50490dfefc7e9684f20/src/pkg/web/www/index.html#L297-L309

其他文件的雷同度更是相当之高，很难让人不怀疑开发者是直接剽窃该项目代码。
在原项目（wg-easy）中使用了CC 4.0协议（Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International
Public License），让我们导航到协议中Section 3的Attribution章节：
https://github.com/wg-easy/wg-easy/blob/c2482f494ae9c7e1acd21db6ab04ee3e91d042da/LICENSE#L237-L281

在此章节中明确要求了二次分发需要署名原作者信息、版权声明、涉及CC协议的声明、涉及免责声明的声明以及原项目URL，但是开发者并没有做到以上任意一点，没有fork、没有原commit历史记录，也没有任何二次分发的说明。这很显然违反了开源协议，综上所述，我不建议上架此应用（我甚至相当推荐上架原项目wg-easy）。

**Comment by Safe3** [comment]
多谢反馈，我们的前端确实参考的是[wg-easy](https://github.com/wg-easy/wg-easy)，但是也仅此而已，主程序完全自主研发和wg-easy没关系，易用性也比wg-easy大大提高，wg-easy需要系统本身安装有wireguard才能使用。我们在软件介绍也说了是plus version of wg-easy

**Comment by kmou424** [comment]
> 多谢反馈，我们的前端确实参考的是[wg-easy](https://github.com/wg-easy/wg-easy)，但是也仅此而已，主程序完全自主研发和wg-easy没关系，易用性也比wg-easy大大提高，wg-easy需要系统本身安装有wireguard才能使用。我们在软件介绍也说了是plus version of wg-easy

是不是参考交给大家说了算，你自己说可没有任何说服力，我不认为有大量完全相同代码段的项目是你所谓的“参考”

**Comment by Safe3** [comment]
Firefly项目主页readme中License部分已经加入前端来自wg-easy，此部分遵循原程序CC4.0的License说明

**Comment by github-zhanglei** [comment]
哈哈！看不清楚背后之间的关系，咱也不敢乱说话！

**Comment by cancanneednews** [comment]
这个东西有代理功能吗？这玩意很敏感，如果1panel上架该应用，他们公司会被警告吧
